File: Programming Basics with Python/06 Nested loops/More exercises/08_Secret_Door's_Lock.py
up_100 = int(input())
up_10 = int(input())
up_1 = int(input())
for number_100 in range(1, up_100 + 1):
    for number_10 in range(2, up_10 + 1):
        if number_10 > 7 or ((number_10 > 2) and (number_10 % 2 == 0)):
            continue
        for number_1 in range(1, up_1 + 1):
            if number_100 % 2 == 0 and number_1 % 2 == 0:
                print(number_100, number_10, number_1)
#-----------------------------------------------------------------------------
up_100 = int(input())
up_10 = int(input())
up_1 = int(input())
for number_100 in range(1, up_100 + 1):
    for number_10 in range(2, up_10 + 1):
        if number_10 > 7:
            continue
        for number_1 in range(1, up_1 + 1):
            if number_100 % 2 == 0 and number_1 % 2 == 0 and ((number_10 == 2) or (number_10 % 2 == 1)):
                print(number_100, number_10, number_1)

File: JS Basics/06 Nested Loops - More Exercises/08. Secret Door's Lock.js
function lock(input) {
    let hundreds = Number(input[0]);
    let tens = Number(input[1]);
    let entity = Number(input[2]);
    for (let up100 = 1; up100 <= hundreds; up100++) {
        for (let up10 = 1; up10 <= tens; up10++) {
            for (let up1 = 1; up1 <= entity; up1++) {
                if ((up1 % 2 == 0 & up100 % 2 == 0) && (((up10 >= 2) && (up10 <= 7)) && (up10 != 4) && (up10 != 6))) {
                    console.log(`${up100} ${up10} ${up1}`);
                }
            }
        }
    }
}
lock([8, 2, 8])
